### PR TITLE
wip(AYC-99): add skill knowledge base use cases and controller

### PR DESCRIPTION
**Pull request by devbydaniel**
> [!NOTE]
> **Medium Risk**
> Adds new skill↔knowledge base association use cases and HTTP endpoints with org/user access checks; mistakes here could expose cross-organization knowledge bases or break skill updates.
> 
> **Overview**
> Adds first-class support for managing *knowledge base assignments on skills*: new use cases to `assign`, `unassign`, and `list` a skill’s knowledge bases, plus a new `SkillKnowledgeBasesController` exposing `POST/DELETE/GET /skills/:skillId/knowledge-bases` routes behind the `Skills` and `KnowledgeBases` feature flags.
> 
> Extends `KnowledgeBaseRepository` with `findByIds` (implemented in `LocalKnowledgeBaseRepository`) to batch-load assigned knowledge bases, and introduces new `SkillErrorCode`s/errors for knowledge-base assignment edge cases (not found, already assigned, wrong org, not assigned). Also adds coverage to ensure `UpdateSkillUseCase` preserves `knowledgeBaseIds` during updates and updates mocks in existing KB tests for the new repository method.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1675edbf1bbfabcb6f26940bb1f504335a4a7a47. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by devbydaniel** [comment]
* **#271** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#270** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/270?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#269** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/269?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#268** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#267** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: ayunis-core-backend/src/domain/skills/application/use-cases/list-skill-knowledge-bases/list-skill-knowledge-bases.use-case.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { ContextService } from 'src/common/context/services/context.service';
+import { UnauthorizedAccessError } from 'src/common/errors/unauthorized-access.error';
+
+@Injectable()
+export class ListSkillKnowledgeBasesUseCase {
+  private readonly logger = new Logger(ListSkillKnowledgeBasesUseCase.name);
+
+  constructor(
+    @Inject(KnowledgeBaseRepository)
+    private readonly knowledgeBaseRepository: KnowledgeBaseRepository,
```
Comment (devbydaniel) [inline ayunis-core-backend/src/domain/skills/application/use-cases/list-skill-knowledge-bases/list-skill-knowledge-bases.use-case.ts:17]: The skill module accesses the knowledge base repository directly here, this is wrong. We need a use case in the knowledge base module to fetch these and use the use case here. Modules only communicate over use cases.

**Comment by devbydaniel** [comment]
### Merge activity

* **Feb 27, 8:03 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/ayunis-core/ayunis-core/269).
* **Feb 27, 8:08 AM UTC**: [Graphite](https://app.graphite.com/github/pr/ayunis-core/ayunis-core/269) rebased this pull request as part of a merge.
* **Feb 27, 8:08 AM UTC**: @devbydaniel merged this pull request with [Graphite](https://app.graphite.com/github/pr/ayunis-core/ayunis-core/269).
